### PR TITLE
Add icon and details only to column view in navigation tree

### LINKED ISSUE #14890: Display column details in navigation bar
It would be very nice to have more information about the columns in the navigation bar, just like SQL Server Management Studio does.

- Primary Key or Foreign Key
- Data Type (int, varchar, date, etc...)
- NULL or NOT NULL

image

### PR DESCRIPTION
**Pull request by saurass**
### Description

Fixes #14890

adding icon and details only to column view in navigation tree

Before submitting pull request, please review the following checklist:

- [x] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [x] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [x] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [x] Every commit has a descriptive commit message.
- [x] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [x] Any new functionality is covered by tests.

### CODE REVIEW

**Review by williamdes — APPROVED** [review]
LGTM for now :)
Tell me when to try your modifications

**Comment by saurass** [comment]
@williamdes 
For now we can see icons in column list along with details -- (data_type, key_type, default_value)
capturepma

**Comment by nunoperalta** [comment]
Looks good! But I'm not sure if the default value should appear there, as it can be a long varchar.

Can we also have "Columns" to auto-expand when we expand the Table?

Meaning, at the moment we have to expand the Table and then expand the Columns. Columns could be auto-expanded by default :)

Thanks very much!

**Comment by saurass** [comment]
Thanks @nunoperalta !

Yeah default value can be removed as it may cause trouble (will remove it).

Yes auto-expansion can be done for sure. @williamdes should this feature be added or it may cause some problem.

**Comment by williamdes** [comment]
> Can we also have "Columns" to auto-expand when we expand the Table?

> Yes auto-expansion can be done for sure.

I would say no, maybe another PR ?


@ibennetch @mauriciofauth Can you please share your thoughts ?

**Comment by saurass** [comment]
@williamdes @ibennetch Should I go further working with this PR?

**Comment by williamdes** [comment]
@saurass I don't know :confused:

**Review by MauricioFauth — APPROVED** [review]

**Comment by mauriciofauth** [comment]
Hello. Thanks for your contribution!

We require a full name for legal compliance reasons. Could you use your full name in the signature tag?
For example: `Signed-off-by: Jane Developer <jane@example.org>`

See [Developer's Certificate of Origin](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO#L39)

File: libraries/classes/Navigation/NavigationTree.php [hunk 1]
```diff
@@ -1192,7 +1192,8 @@ private function _renderNode($node, $recursive, $class = '')
                 } else {
                     $retval .= "<a class='hover_show_full$linkClass' href='$link'";
                     $retval .= " title='$title'>";
-                    $retval .= htmlspecialchars($node->real_name);
+                    $retval .= htmlspecialchars(isset($node->disp_name) ?
+                                    $node->disp_name : $node->real_name);
```
Comment (mauriciofauth) [inline libraries/classes/Navigation/NavigationTree.php]: You can use `htmlspecialchars($node->disp_name ?? $node->real_name)` here.

**Comment by nunoperalta** [comment]
@saurass - did you end up not removing the default value? I think that can be problematic, if the default value is too large or contains special symbols, etc...
Thank you.

**Comment by mauriciofauth** [comment]
I agree. It's also possible to truncate the default value instead of removing it. And the default value needs to be escaped.

File: libraries/classes/Navigation/NavigationTree.php (hunk 1 above)
Comment (saurass) [reply libraries/classes/Navigation/NavigationTree.php]: Yeah Sure

**Comment by saurass** [comment]
> Hello. Thanks for your contribution!
> 
> We require a full name for legal compliance reasons. Could you use your full name in the signature tag?
> For example: `Signed-off-by: Jane Developer <jane@example.org>`
> 
> See [Developer's Certificate of Origin](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO#L39)

Ya sure I just amended the commit messages

**Comment by saurass** [comment]
> I agree. It's also possible to truncate the default value instead of removing it. And the default value needs to be escaped.

Okay working on it. Thanks : )

**Comment by saurass** [comment]
> And the default value needs to be escaped.

@mauriciofauth  What escape, sorry I didn't get this one.

**Comment by nunoperalta** [comment]
I think he meant to use something like htmlspecialchars, so that HTML tags aren't parsed and cause UI issues.

**Comment by saurass** [comment]
> I think he meant to use something like htmlspecialchars, so that HTML tags aren't parsed and cause UI issues.

So should I strip HTML tags or show the HTML tags as well?

**Comment by nunoperalta** [comment]
I think show HTML tags, in plain text.

**Comment by nunoperalta** [comment]
Are you able to add whether column is NULL or NOT NULL as well? :)

datatype, keytype, nulltype, default

**Comment by saurass** [comment]
> I think show HTML tags, in plain text.

That is showing at the moment

**Comment by saurass** [comment]
> Are you able to add whether column is NULL or NOT NULL as well? :)
> 
> datatype, keytype, nulltype, default

Okay, lemme have a look

**Comment by saurass** [comment]
@sudo-bot Deploy pr

**Comment by nunoperalta** [comment]
> In case the key of column is primary or index, then only key_type is shown

I think you mean key_type is only shown if the column is a key (took a while to understand!) - that's perfect.

But, are you not planning to show whether the column allows NULLs then?

Thank you for your time!

**Comment by saurass** [comment]
Just added the nullable feature.

@williamdes @nunoperalta, please have a look. : )

**Comment by saurass** [comment]
@sudo-bot Deploy pr

**Comment by nunoperalta** [comment]
@saurass You are a star! Thank you very much.

**Comment by saurass** [comment]
Thanks @nunoperalta : )

**Review by MauricioFauth — APPROVED** [review]

**Comment by mauriciofauth** [comment]
Merged, thanks for your contribution!

**Comment by saurass** [comment]
Thank you @mauriciofauth

**Pull request merged**